custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/service/LineOfCreditWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+            final Locale locale = Locale.forLanguageTag(localeString);
+            final LocalDate startDate = command.localDateValueOfParameterNamed("startDate");
+            final LocalDate endDate = command.localDateValueOfParameterNamed("endDate");
+
+
+            final LineOfCredit lineOfCredit = new LineOfCredit(client, name, productType, maximumAmount, startDate, endDate);
+            final LineOfCredit savedLineOfCredit = this.lineOfCreditRepository.save(lineOfCredit);
+
+
+            // Create a map with the line of credit data
+            final Map<String, Object> changes = new HashMap<>();
+            changes.put("resourceId", savedLineOfCredit.getId());
+            changes.put("clientId", savedLineOfCredit.getClient().getId());
+            changes.put("name", savedLineOfCredit.getName());
+            changes.put("productType", savedLineOfCredit.getProductType());
+            changes.put("maximumAmount", savedLineOfCredit.getMaximumAmount());
+            changes.put("availableBalance", savedLineOfCredit.getAvailableBalance());
+            changes.put("consumedAmount", savedLineOfCredit.getConsumedAmount());
+            changes.put("activationStatus", savedLineOfCredit.getActivationStatus().name());
+            changes.put("startDate", savedLineOfCredit.getStartDate());
+            changes.put("endDate", savedLineOfCredit.getEndDate());
+
+            return new CommandProcessingResultBuilder().withEntityId(savedLineOfCredit.getId()).with(changes).build();
+
+        } catch (final PlatformApiDataValidationException e) {
+            // Re-throw validation exceptions as-is
+            log.error("Validation error occurred: {}", e.getMessage());
+            throw e;
+        } catch (final Exception e) {
+            log.error("Error occurred while creating line of credit", e);
+            throw new PlatformApiDataValidationException("error.msg.line.of.credit.creation.failed", "Line of credit creation failed",
+                    e.getMessage(), e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult updateLineOfCredit(Long lineOfCreditId, JsonCommand command) {
+        try {
+            this.dataValidator.validateForUpdate(command.json());
+
+            final LineOfCredit lineOfCredit = this.lineOfCreditRepository.findById(lineOfCreditId)
+                    .orElseThrow(() -> new PlatformApiDataValidationException("error.msg.line.of.credit.not.found",
+                            "Line of credit not found", "lineOfCreditId"));
+
+            final Map<String, Object> changes = lineOfCredit.update(command);
+
+            if (!changes.isEmpty()) {
+                this.lineOfCreditRepository.save(lineOfCredit);
+            }
+
+
+            // Create a map with the updated line of credit data
+            final Map<String, Object> responseData = new HashMap<>();
+            responseData.put("resourceId", lineOfCredit.getId());
```

Comment:
This is also wrong, if you follow the update logic in fineract, it will always check if a value has changed and it will only return what has actually changed in this payload. You are returning everything.